lib/util.js

```diff
@@ -478,7 +478,20 @@ function formatValue(ctx, value, recurseTimes) {
   if (ctx.showHidden) {
     keys = Object.getOwnPropertyNames(value);
   } else {
-    keys = Object.keys(value);
+    // This might throw if `value` is a Module Namespace Object from an
+    // unevaluated module, but we don't want to perform the actual type
+    // check because it's expensive.
+    try {
+      keys = Object.keys(value);
+    } catch (err) {
+      if (err instanceof ReferenceError &&
+          types.isModuleNamespaceObject(value)) {
+        keys = Object.getOwnPropertyNames(value);
+      } else {
+        throw err;
```

Comment:
☝️ the `Object.getOwnPropertyNames(value)` fallback should filter non-enumerable to be equiv to `keys`

Reply:
I suppose you can't have non-enumerable keys on a module namespace?

Reply:
@jdalton until the spec changes we have no way to check which ones are enumerable, and what @targos said

Reply:
i believe the only keys on a module namespace will ever (atm) be enumerable strings for the exports, and `Symbol.toStringTag`.